### PR TITLE
2674888 Added query string and populate add to cart

### PR DESCRIPTION
**Pull request by haringsrob**
We might want to improve by adding a constant for "dv" so that if we ever change the tactic, it will be an easy task.

### CODE REVIEW

File: modules/cart/src/Form/AddToCartForm.php [hunk 1]
```diff
@@ -81,6 +81,13 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       $form['#access'] = FALSE;
     }
 
+    // If our default variation is set, we alter the add to cart form.
+    if ($default_variation = \Drupal::request()->query->get('dv')) {
+      if (is_numeric($default_variation)) {
+        $form['purchased_entity']['widget'][0]['attributes']['field_variation']['#default_value'] = $default_variation;
```
Comment (mglaman) [inline modules/cart/src/Form/AddToCartForm.php]: Should we check if this is a valid product variation identifier for the given product, too?

Comment (mglaman) [inline modules/cart/src/Form/AddToCartForm.php]: Also, if this is given, we could set it earlier and bypass other logic, making it a bit more performant

Comment (haringsrob) [inline modules/cart/src/Form/AddToCartForm.php]: I was thinking about validation as well. But how will we handle it then? Set a message? Should we allow this message to be altered?

I agree on doing this at an earlier stage, but I will need a suggestion where? viewElements?

Comment (mglaman) [inline modules/cart/src/Form/AddToCartForm.php]: You'd want to move this to the new field widget, for product attributes. Should be able to check before it invokes its methods to discover the variation to be selected by default

**Pull request closed without merging**